app/scripts/controller-init/test/utils.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  BaseRestrictedControllerMessenger,
+  ControllerInitRequest,
+  ControllerPersistedState,
+} from '../types';
+
+export const CHAIN_ID_MOCK = '0x123';
+
+export function buildControllerInitRequestMock(): jest.Mocked<
+  Omit<
+    ControllerInitRequest<
+      BaseRestrictedControllerMessenger,
+      BaseRestrictedControllerMessenger
+    >,
+    'controllerMessenger' | 'initMessenger'
+  >
+> {
+  return {
+    getController: jest.fn(),
+    getFlatState: jest.fn(),
+    getGlobalChainId: jest.fn().mockReturnValue(CHAIN_ID_MOCK),
+    getPermittedAccounts: jest.fn(),
+    getProvider: jest.fn(),
+    getTransactionMetricsRequest: jest.fn(),
+    persistedState: {} as ControllerPersistedState,
```

Comment:
Nit: Unnecessary cast, `{}` is already compatible with this type since `Partial` was added
```suggestion
    persistedState: {},
```

Reply:
Done.